netmap/policy.go

```diff
@@ -30,6 +30,46 @@ type PlacementPolicy struct {
 	replicas []netmap.Replica
 }
 
+func copyFilter(f netmap.Filter) netmap.Filter {
+	var filter netmap.Filter
+
+	filter.SetName(f.GetName())
+	filter.SetKey(f.GetKey())
+	filter.SetOp(f.GetOp())
+	filter.SetValue(f.GetValue())
+
+	filters := make([]netmap.Filter, len(f.GetFilters()))
+	for i, internalFilter := range f.GetFilters() {
+		filters[i] = copyFilter(internalFilter)
+	}
+
+	filter.SetFilters(filters)
+
+	return filter
+}
+
+// CopyTo writes deep copy of the [PlacementPolicy] to dst.
+func (p PlacementPolicy) CopyTo(dst *PlacementPolicy) {
+	if dst == nil {
```

Comment:
passing nil is absolute nonsense, so we should panic

Reply:
1st example is a programmer's mistake, so panic is ok. Otherwise method doesn't do its functionality (nothing copied, no failures)

Reply:
I'm against panic in this case too. We just check the input value. In this exact case, `nil` is not a reason to `panic` manually.

I don't mind to remove this condition at all. If someone passes nil, it would be its responsibility. Anyway, it has a panic because nil reference error.

Reply:
I'd remove this `nil` check, it's passing `nil` to https://pkg.go.dev/net/http#Client.Do

Reply:
Removed